### PR TITLE
allow for bool in _parameters again, integer -> int, character -> string

### PR DESCRIPTION
**Pull request by zingale**
many int parameters were changed to bool -- this will help when we go
all-in on a struct

also replace all the X.dY with C++ style

### CODE REVIEW

**Comment by zingale** [comment]
tests pass: http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2024-04-29-001/index.html

File: opacity/rad_power_law/_parameters [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
 # Frequency exponent (scattering)
-scatter_exp_p                real            0.0d0
+scatter_exp_p                real            0.0e0
 
 # Opacity floor
-kappa_floor                  real            0.0d0
+kappa_floor                  real            0.0e0
 
 # Temperature floor
 rad_temp_floor               real            1.d-10
```
Comment (yut23) [inline opacity/rad_power_law/_parameters:43]:
```suggestion
rad_temp_floor               real            1.e-10
```

File: unit_test/burn_cell/_parameters [hunk 2]
```diff
@@ -12,7 +12,7 @@ tmax          real       1.e-2
 tfirst        real       0.0
 
 # number of steps (logarithmically spaced)
-nsteps        integer    100
+nsteps        int    100
 
 density       real       1.d7
```
Comment (yut23) [inline unit_test/burn_cell/_parameters:17]:
```suggestion
density       real       1.e7
```

File: integration/_parameters [hunk 3]
```diff
@@ -6,103 +6,102 @@
...
 # flag for turning on the use of number densities for all species
-use_number_densities     integer     0
+use_number_densities     bool     0
 
 # flag for tuning on the subtraction of internal energy
-subtract_internal_energy   integer     1
+subtract_internal_energy   bool     1
 
 # SDC iteration tolerance adjustment factor
 sdc_burn_tol_factor          real         1.d0
```
Comment (yut23) [inline integration/_parameters:88]:
```suggestion
sdc_burn_tol_factor          real         1.e0
```

File: unit_test/burn_cell/_parameters [hunk 4]
```diff
@@ -12,9 +12,9 @@ tmax          real       1.e-2
 tfirst        real       0.0
 
 # number of steps (logarithmically spaced)
-nsteps        integer    100
+nsteps        int    100
 
-density       real       1.d7
+density       real       1.e7
 
 temperature   real       3.d9
```
Comment (yut23) [inline unit_test/burn_cell/_parameters:19]:
```suggestion
temperature   real       3.e9
```

File: unit_test/burn_cell/_parameters [hunk 5]
```diff
@@ -12,10 +12,10 @@ tmax          real       1.e-2
...
-nsteps        integer    100
+nsteps        int    100
 
-density       real       1.d7
+density       real       1.e7
 
-temperature   real       3.d9
+temperature   real       3.e9
 
 skip_initial_normalization    int    0
```
Comment (yut23) [inline unit_test/burn_cell/_parameters:21]:
```suggestion
skip_initial_normalization    bool    0
```

File: unit_test/burn_cell_sdc/_parameters [hunk 6]
```diff
@@ -10,7 +10,7 @@ tmax          real       1.e-2
 tfirst        real       0.0
 
 # number of steps (logarithmically spaced)
-nsteps        integer    100
+nsteps        int        100
 
 # do we recompute the aux quantities? or do we take them as given in the inputs?
 recompute_aux integer    0
```
Comment (yut23) [inline unit_test/burn_cell_sdc/_parameters:16]:
```suggestion
recompute_aux bool       0
```

**Review by yut23 — COMMENTED** [review]
`integration/BackwardEuler/_parameters` and `integration/QSS/_parameters` need integer -> int replacements; `unit_test/burn_cell_primordial_chem/_parameters` needs character -> string. Looks good otherwise

**Comment by zingale** [comment]
updated the remaining.  Let's see if the tests pass...

**Comment by zingale** [comment]
okay, I think that this is all set now

**Review by yut23 — APPROVED** [review]

**Pull request merged**